src/rnd.cpp

```diff
@@ -58,10 +58,10 @@
 //  Has a full period of 2^31 - 1.
 //  Returns integers in the range 1 to 2^31-1.
 
-#define RNG_M 2147483647L // m = 2^31 - 1
-#define RNG_A 16807L
-#define RNG_Q 127773L // m div a
-#define RNG_R 2836L   // m mod a
+constexpr long RNG_M = 0b10000000000000000000000000000000L - 1; // m = 2^31 - 1
```

Comment:
A binary literal is likely not ideal for so such long numbers. Just to use the number is not as easy to understand as an expression in this case, something like (1 << 31 )- 1, I think.